app/models/manageiq/providers/redhat/infra_manager/vm/operations/snapshot.rb

```diff
@@ -20,15 +20,15 @@ def raw_create_snapshot(_name, desc, memory)
   end
 
   def raw_remove_snapshot(snapshot_id)
-    snapshot = snapshots.find_by_id(snapshot_id)
+    snapshot = snapshots.find_by(:id => snapshot_id)
```

Comment:
This `find_by_id` is mocked up in `snapshot_spec.rb`.

The `snapshot_spec.rb` needs to be amended to mock-up `find_by` instead.